serialization/class_definition.go

```diff
@@ -0,0 +1,177 @@
+	cd.addNewFieldDefinition(fieldName, TypeFloat32)
+}
+
+func (cd *ClassDefinition) AddFloat64Field(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeFloat64)
+}
+
+func (cd *ClassDefinition) AddStringField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeString)
+}
+
+func (cd *ClassDefinition) AddPortableField(fieldName string, def *ClassDefinition) error {
+	if def.ClassID == 0 {
+		return hzerrors.NewHazelcastIllegalArgumentError("Portable class ID cannot be zero", nil)
+	}
+	cd.AddField(NewFieldDefinition(int32(len(cd.Fields)), fieldName, TypePortable, def.FactoryID, def.ClassID, cd.Version))
+	return nil
+}
+
+func (cd *ClassDefinition) addNewFieldDefinition(fieldName string, fieldType int32) {
+	cd.AddField(NewFieldDefinition(int32(len(cd.Fields)), fieldName, fieldType, 0, 0, cd.Version))
+}
+
+func (cd *ClassDefinition) AddByteArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeByteArray)
+}
+
+func (cd *ClassDefinition) AddBoolArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeBoolArray)
+}
+
+func (cd *ClassDefinition) AddInt16ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeInt16Array)
+}
+
+func (cd *ClassDefinition) AddUInt16ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeUInt16Array)
+}
+
+func (cd *ClassDefinition) AddInt32ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeInt32Array)
+}
+
+func (cd *ClassDefinition) AddInt64ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeInt64Array)
+}
+
+func (cd *ClassDefinition) AddFloat32ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeFloat32Array)
+}
+
+func (cd *ClassDefinition) AddFloat64ArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeFloat64Array)
+}
+
+func (cd *ClassDefinition) AddPortableArrayField(fieldName string, def *ClassDefinition) error {
+	if def.ClassID == 0 {
+		return hzerrors.NewHazelcastIllegalArgumentError("Portable class ID cannot be zero", nil)
+	}
+	cd.AddField(NewFieldDefinition(int32(len(cd.Fields)), fieldName, TypePortableArray, def.FactoryID, def.ClassID, cd.Version))
+	return nil
+}
+
+func (cd *ClassDefinition) AddStringArrayField(fieldName string) {
+	cd.addNewFieldDefinition(fieldName, TypeStringArray)
+}
+
+const (
+	TypePortable int32 = iota
+	TypeByte
+	TypeBool
+	TypeUint16
+	TypeInt16
+	TypeInt32
+	TypeInt64
+	TypeFloat32
+	TypeFloat64
+	TypeString
+	TypePortableArray
+	TypeByteArray
+	TypeBoolArray
+	TypeUInt16Array
+	TypeInt16Array
+	TypeInt32Array
+	TypeInt64Array
+	TypeFloat32Array
+	TypeFloat64Array
+	TypeStringArray
+)
+
+// FieldDefinition defines name, type, index of a field.
+type FieldDefinition struct {
+	Name      string
+	Index     int32
+	Type      int32
+	FactoryID int32
+	ClassID   int32
+	Version   int32
+}
+
+func NewFieldDefinition(index int32, fieldName string, fieldType int32, factoryID int32,
```

Comment:
I think users shouldn't be able to create field definitions themselves

Reply:
I've unexported `NewFieldDefinition` at: https://github.com/hazelcast/hazelcast-go-client/pull/569/commits/886c827b7a16d536cd96df45676e377f730bc0e0

Reply:
One more newby-Go-reviewer question :smile: 

Can I use the function defined on the internal package here and remove the `newFieldDefinition` from `serialization/class_definition.go` to avoid code duplication?

Reply:
Np at all, thanks a lot for the review! :)

Internal `serialization` package depends on the public one, so it's not possible to put `NewFieldDefinition` into `internal/serialization`. It's possible to move it to another package, or a new internal package, but I think that would remove it from the package it is used, so I don't like that idea much.